From 911e282377c9d446144fb4409d4090416db540f3 Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Wed, 21 May 2014 03:14:20 +0530
Subject: [PATCH 04/24] ASoC: Intel: platform: Add mux and debug control
 support

mux control and debug control support to be parsed using
DFW binary is added.

Issue: GMINL-3680

Change-Id: If8970fe99d0c4791918fadce3ddcd1f5429e3342
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 10 ++++++----
 sound/soc/intel/platform-libs/sst_vendor.h       | 18 ++++++++++++++++++
 2 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 8a388ed..bb6b5f2 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -2024,6 +2024,12 @@ const struct snd_soc_fw_kcontrol_ops control_ops[] = {
 	{SOC_CONTROL_IO_SST_ALGO_BYPASS, sst_algo_control_get, sst_algo_control_set, snd_soc_info_bool_ext},
 	{SOC_CONTROL_IO_SST_MIX, sst_mix_get, sst_mix_put, snd_soc_info_volsw},
 	{SOC_CONTROL_IO_SST_MUX, sst_mux_get, sst_mux_put, snd_soc_info_enum_double},
+	{SOC_CONTROL_IO_SST_BYTE, sst_byte_control_get, sst_byte_control_set,
+						snd_soc_info_enum_double},
+	{SOC_CONTROL_IO_SST_MODE, sst_mode_get, sst_mode_put,
+						snd_soc_info_enum_double},
+	/*{SOC_CONTROL_IO_SST_VOICE_MODE, sst_mode_get, sst_voice_mode_put,
+						snd_soc_info_enum_double},*/
 };
 
 const struct snd_soc_fw_widget_events sst_widget_ops[] = {
@@ -2290,10 +2296,6 @@ int sst_dsp_init_v2_dpcm_dfw(struct snd_soc_platform *platform)
 	}
 	snd_soc_add_platform_controls(platform, sst_slot_controls,
 			ARRAY_SIZE(sst_slot_controls));
-	snd_soc_add_platform_controls(platform, sst_mux_controls,
-			ARRAY_SIZE(sst_mux_controls));
-	snd_soc_add_platform_controls(platform, sst_debug_controls,
-			ARRAY_SIZE(sst_debug_controls));
 	snd_soc_add_platform_controls(platform, sst_vad_enroll,
 			ARRAY_SIZE(sst_vad_enroll));
 
diff --git a/sound/soc/intel/platform-libs/sst_vendor.h b/sound/soc/intel/platform-libs/sst_vendor.h
index da8ed8d..b6c8f84 100644
--- a/sound/soc/intel/platform-libs/sst_vendor.h
+++ b/sound/soc/intel/platform-libs/sst_vendor.h
@@ -30,6 +30,9 @@
 #define SOC_CONTROL_TYPE_SST_ALGO_BYPASS	103
 #define SOC_CONTROL_TYPE_SST_MUX			104
 #define SOC_CONTROL_TYPE_SST_MIX			106
+#define SOC_CONTROL_TYPE_SST_BYTE           108
+#define SOC_CONTROL_TYPE_SST_MODE           109
+#define SOC_CONTROL_TYPE_SST_VOICE_MODE           110
 
 /* REVISIT: Define sst kcontrol index */
 #define SOC_CONTROL_IO_SST_GAIN\
@@ -63,6 +66,21 @@
 		SOC_CONTROL_TYPE_SST_MUX, \
 		SOC_CONTROL_TYPE_SST_MUX)
 
+#define SOC_CONTROL_IO_SST_BYTE\
+	SOC_CONTROL_ID(SOC_CONTROL_TYPE_SST_BYTE, \
+		SOC_CONTROL_TYPE_SST_BYTE, \
+		SOC_CONTROL_TYPE_SST_BYTE)
+
+#define SOC_CONTROL_IO_SST_MODE\
+	SOC_CONTROL_ID(SOC_CONTROL_TYPE_SST_MODE, \
+		SOC_CONTROL_TYPE_SST_MODE, \
+		SOC_CONTROL_TYPE_SST_MODE)
+
+#define SOC_CONTROL_IO_SST_VOICE_MODE\
+	SOC_CONTROL_ID(SOC_CONTROL_TYPE_SST_VOICE_MODE, \
+		SOC_CONTROL_TYPE_SST_VOICE_MODE, \
+		SOC_CONTROL_TYPE_SST_VOICE_MODE)
+
 #define SND_SOC_GAIN_CONTROL_NAME 44
 
 /* Event types goes here */
-- 
1.9.1

